pkg/oauth2ext/devicecode/devicecode.go

```diff
@@ -125,10 +125,18 @@ func (c *Config) DeviceCodeExchange(ctx context.Context, deviceCode string) (*oa
 			Body:     body,
 		}
 	default:
+		// TODO accept application/x-www-form-urlencoded and text/plain responses in addition to json
```

Comment:
You don't need to make these changes here -- just change the mocks so that they also report a `StatusCode` instead of just `Status` and the handler above will decode them correctly. We shouldn't be getting errors with 200 responses.

Reply:
Hmm. Then it's coming back as a `oauth2.ResponseError` wrapped in a `errmap.mappedError`, but I can't access as a `mappedError` because that's a private interface and none of the other `errmap` public interfaces implement `error` either. Not sure what to do here to get at the `oauth2.ResponseError`...

Reply:
It supports the new Go 1.13+ error wrapping, so you can use `errors.As()` to get it, or look at some of the helper methods in `errmap`/`errmark` as they might give the level of information you need (e.g., is this a transient error?).

Reply:
Cool! I didn't know about that.

Reply:
Oh, and this code comment is specifically about https://github.com/golang/oauth2/blob/cd4f82c27b84ec2516d3c38a5b5acaf8823e2702/internal/token.go#L249-L280

Reply:
I believe that those response media types never made it through to the RFC 6749 version of OAuth 2, so they wouldn't apply to device flow.